### ISSUE TITLE
SQL non idempotent commands aren't transactional when use remote protocol [moved]

### ISSUE THREAD

**Issue by lvca** [orientechnologies/orientdb#181]
This is [Issue 181](http://code.google.com/p/orient/issues/detail?id=181) moved from a Google Code project.
Added by 2010-12-20T18:03:06.000Z by [l.garu...@gmail.com](http://code.google.com/u/111607061083712272202/).
Please review that bug for more context and additional comments, but update this bug.

Original labels: Type-Defect, Priority-Low, v1.4.0
### Original description

```
SQL operation like:
- Insert
- Update
- Delete

are non transactionals since they act at server side if you're using the remote protocol.
```

**Comment by lvca** [comment]
A solution to this issue could be supporting a new TX_FLUSH command, where the client flushes the TX to the server. Every time a new command is executed, the TX is flushed to the server. 

We could also create a new setting to allow the client to flush to the server at regular basis avoiding to use too much memory on the client.

## How to implement it?
Just get the first part of TX_COMMIT command

**Comment by laa** [comment]
Assigned to @tglman

**Comment by lvca** [comment]
This fixes also https://github.com/orientechnologies/orientdb/issues/2875

**Comment by tglman** [comment]
Hi,
Implemented in 3.0, closing
